skimage/measure/_structural_similarity.py

```diff
@@ -48,6 +155,69 @@ def structural_similarity(X, Y, win_size=7,
     if not X.shape == Y.shape:
         raise ValueError('Input images must have the same dimensions.')
 
+    if image_content_weighting and gradient:
+        raise ValueError(
+            "gradient not implemented for image content weighted case")
+
+    # default treats 3D arrays with shape[2] == 3 as multichannel
```

Comment:
In the past, the devs as a group decided against using this kind of magic to determine whether an image was multichannel. See the discussion [here](https://github.com/scikit-image/scikit-image/pull/719#issuecomment-23727951). See the treatment of multichannel in [slic](https://github.com/scikit-image/scikit-image/blob/master/skimage/segmentation/slic_superpixels.py#L115) for a template. (Although, the default of `True` in SLIC might need to be changed to `False` here because of different typical use-cases.)

As you might see in the discussion linked above, I actually quite like this logic, but @JDWarner's argument was pretty convincing and I think consistency across the package is important.